noir-projects/aztec-nr/aztec/src/oracle/get_plume_proof.nr

```diff
@@ -0,0 +1,216 @@
+ *       have to inject data into the circuit via function args or via 
+ *       capsules.
+ * - Can I pass an address_pk instead of a nullifier_pk to this function?
+ *     - Technically, yes, although:
+ *         - It muddles the trust assumptions of what each key should be used
+ *           for. It would require usage of the ivsk, and that key was 
+ *           intended for decryption rather than nullification.
+ *           For some wallets, the nsk and ivsk might live in the same place,
+ *           in which case perhaps it's fine to use that key. But actually,
+ *           here we're in the context of an app contract. And an app contract
+ *           really ought to seek to be compatible with all wallets.
+ *     - The Aztec-Labs PXE's version of this oracle will not work with an 
+ *       address_sk, currently.
+ *             
+ */
+pub fn get_plume_proof_unsafe<let N: u32>(msg: [Field; N], pk_m: Point) -> PlumeProof<N> {
+    // Notice that the oracle call doesn't return a PlumeProof. We want to save
+    // on constraints, and it's always dangerous injecting more witness data
+    // than necessary.
+    // The oracle returns the bare minimum amount of data, and then (here in
+    // _constrained_-land) we add existing witness data (msg and pk_m) to
+    // create the returned PlumeProof.
+
+    // Safety: Unsafe because it's an oracle call.
+    let PlumeProofOracleReturnType { _nullifier_point, _A2, _B2, _s } =
+        unsafe { get_plume_proof_inner(msg, pk_m) };
+
+    let nullifier_point =
+        Point { x: _nullifier_point[0], y: _nullifier_point[1], is_infinite: false };
+
+    let A2 = Point { x: _A2[0], y: _A2[1], is_infinite: false };
+
+    let B2 = Point { x: _B2[0], y: _B2[1], is_infinite: false };
+
+    // TODO: construction in this way might not be safe, because
+    let s = Scalar::new(_s[0], _s[1]);
+
+    PlumeProof { msg, pk: pk_m, nullifier_point, A2, B2, s }
+}
+
+mod test {
+    use crate::{
+        nullifier::plume::{
+            compute_plume_proof_in_tests::{compute_plume_proof_in_tests, ScalarWithOps},
+            utils::scalar_mul,
+            verify_plume_proof::verify_plume_proof,
+        },
+        test::helpers::test_environment::TestEnvironment,
+    };
+    use super::get_plume_proof_unsafe;
+    use std::embedded_curve_ops::{
+        EmbeddedCurvePoint as Point, EmbeddedCurveScalar as Scalar,
+        FIXED_BASE_SCALAR_MUL_GENERATOR as G,
+    };
+
+    #[test]
+    unconstrained fn test_compute_plume_nullifier() {
+        // We need to establish a test environment, so that the keypairs we use
+        // match those of the TXE.
+        let mut env = TestEnvironment::new();
+        let new_account_secret = 1;
+
+        // It's unsafe to access the master secret keys within a circuit.
+        // It's acceptable here, because it's a test env.
+        let unsafe_account = env.unsafe_create_account(new_account_secret);
```

Comment:
So I introduced a new cheatcode oracle `unsafe_create_account`, just to get access to the nullifier secret key, so that I could compare a computation in noir-land against a computation in typescript-land (both computations required a secret key). This is a controversial oracle, and perhaps these equality tests should instead be performed by comparing hard-coded test fixtures in ts and nr.